x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java

```diff
@@ -93,7 +97,13 @@ private void cleanUpAnalytics() {
         for (DataFrameAnalyticsConfig config : analytics) {
             try {
                 assertThat(deleteAnalytics(config.getId()).isAcknowledged(), is(true));
-                assertThat(searchStoredProgress(config.getId()).getHits().getTotalHits().value, equalTo(0L));
+                if (searchStoredProgress(config.getId()).getHits().getTotalHits().value > 0) {
```

Comment:
Could you explain how we could have leftover progress here?

Reply:
@dimitris-athanasiou the overall change only partially handles the race condition window.

- force close a job
- persist progress
- delete job (deletes previous progress)
- job finally fires onCancelled()
- job sees that there was no previous progress persisted
- persists the new progress

Reply:
OK, I think we can then remove the assertion here entirely. I'll add a test dedicated to deleting a job and checking things like the stored progress doc was cleaned up in a separate PR.